amivapi/validation.py

```diff
@@ -121,3 +122,22 @@ def _validate_filetype(self, filetype, field, value):
         if not(t in filetype):
             self._error(field, "filetype '%s' not supported, has to be in: "
                         "%s" % (t, filetype))
+
+    def _validate_session_younger_than(self, threshold_time, field, value):
+        """Validation of the used token for special fields
+
+        Validates if the session is not older than threshold_time
+
+        Except admins, they can ignore this
+
+        Args:
+            threshold_time (int): threshold to to compare with in seconds
+            field (string): field name.
+            value: field value.
+        """
+        time_created = g.current_session['_created'].replace(tzinfo=None)
+        time_now = datetime.now()
+
+        if threshold_time > 0 and time_now - time_created > threshold_time:
```

Comment:
I am a bit worried about the timezones here. datetime.now() returns a local time, but as far as I remember we want UTC everywhere (and even enforce UTC in the external API).
How about:

```
time_created = g.current_session['_created']
time_now = datetime.now(timezone.utc)
```

Then the comparison should be safe.

Reply:
This is a great idea as far as the times saved in the database are also in UTC.

Reply:
The easiest way would be using`datetime.utcnow()`, I guess

Reply:
datetime.utcnow does not have a timezone attached. That's why I prefer the other way ;)